jsHelper/spicetifyWrapper.js

```diff
@@ -921,6 +934,23 @@ Spicetify.Events = (() => {
 		playerState.cache = playerState.current;
 	});
 
+	(function waitProductStateAPI() {
+		if (!Spicetify.Platform?.UserAPI) {
+			setTimeout(waitUserAPI, 100);
+			return;
+		}
+
+		const productState = Spicetify.Platform.UserAPI?._product_state || Spicetify.Platform.UserAPI?._product_state_service;
```

Comment:
The `if (!Spicetify.Platform?.UserAPI) { /* ... */; return }` check prevents `Spicetify.Platform.UserAPI` from ever being null or undefined, so the optional chaining is not needed.

Reply:
Not valid. The prop can be undefined so it's better to stay like this

Reply:
`a?.b` is equivalent to `a === null || a === undefined ? undefined : a.b`, b is irrelevant from the operator's perspective. The fact that `_product_state` and `_product_state_service` can be undefined changes nothing.

Reply:
right... mb